### DISCUSSION TITLE
UEFI network boot of laptop with USB Ethernet using iPXE?

### DISCUSSION THREAD

**Discussion by LRSFC-DanJ** [ipxe/ipxe#1000] [Q&A] [answered]
Hello,

At my workplace, some of our laptops don't have built-in Ethernet adaptors and don't natively support booting from a USB Ethernet adaptor.

Is there a way for us to use iPXE on a USB stick to somehow chain load the existing PXE image on our existing PXE server using the USB Ethernet adaptor? I.e. for iPXE to initialise the USB Ethernet adaptor and perform the DHCP request and act accordingly as the DHCP PXE options dictate?

We are also on Windows, we don't have access to any Linux machines so we would need a method using pre-existing binary files. Is this possible?

We tried downloading ipxe.efi from https://boot.ipxe.org/ipxe.efi and renaming it to bootx64.efi and putting it in an \EFI\BOOT folder on a USB stick and while this did run, it does not appear to be able to detect the USB Ethernet adaptor, what are we missing?

**Comment by NiKiZe** [comment, marked as answer]
This is almost identical to https://github.com/ipxe/ipxe/discussions/848
So yes there is some native drivers for USB adapters.

You will need to build this yourself. You can however do this just fine under WSL2 in windows with a basic Ubuntu or other basic distro.

This will use iPXE extensions of network boot. But should be compatible in most casese.
